### PR TITLE
Add PropertyHook

### PR DESCRIPTION
**Pull request by kuraga**
...in addition to `AttributeHook`

### CODE REVIEW

File: virtual-hyperscript/test/property-hook.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    assert.equal(elem.myprop, 'second value')
+
+    patches = diff(third, fourth)
+    patch(elem, patches)
+    assert.equal(elem.myprop, undefined)
+
+    assert.end()
+})
+
+test("sets the property if previous value was not an AttributeHook", function (assert) {
```
Comment (neonstalwart) [inline virtual-hyperscript/test/property-hook.js]: s/AttributeHook/PropertyHook/

File: virtual-hyperscript/test/property-hook.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    assert.equal(elem.myprop, blankAttributeNS())
+
+    patches = diff(first, second)
+    patch(elem, patches)
+    assert.equal(elem.myprop, 'the value')
+
+    assert.end()
+})
+
+test("removes the property if next value is not an AttributeHook", function (assert) {
```
Comment (neonstalwart) [inline virtual-hyperscript/test/property-hook.js]: s/AttributeHook/PropertyHook/

**Comment by neonstalwart** [comment]
why do we need this?  ...and why not just update SoftSetHook since this is what it already does?

**Comment by kuraga** [comment]
In addition to `AttributeHook`. The same but for properties.

**Comment by neonstalwart** [comment]
and why not just use (or open a PR to update and maybe rename) https://github.com/Matt-Esch/virtual-dom/blob/88534bbd34c5befbd1689fc71e2373c9145a9807/virtual-hyperscript/hooks/soft-set-hook.js?

**Comment by kuraga** [comment]
@neonstalwart Wow! It's really the same! I saw the name "`setSoft`" and even didn't thought about the same functionality!!!

**Comment by kuraga** [comment]
Sure we should save one only. But `PropertyHook` (as same as  `AttributeHook`) checks `prev && prev.type === 'PropertyHook'`. But `SoftSetHook` doesn't.

**Comment by neonstalwart** [comment]
> (or open a PR to update and maybe rename)

**Comment by kuraga** [comment]
Yeah, I'm talking about:
1) update or check is needless? (I think - update),
2) rename?

**Comment by neonstalwart** [comment]
yeah, i think we should improve `SoftSetHook`.  

if you think `PropertyHook` is a better name for it, then leave the code in virtual-hyperscript/hooks/property-hook.js and update virtual-hyperscript/hooks/soft-set-hook.js to be something like:

```js
// DEPRECATED: use property-hook instead
module.exports = require('./property-hook');
```

**Comment by kuraga** [comment]
Thanks. Closed in favor of #242

**Pull request closed without merging**